.github/workflows/test.yml

```diff
@@ -0,0 +1,41 @@
+name: test
+on: [push, pull_request]
+env:
+  SERVER_AUTH_TYPE: basic
+  SERVER_URL: http://127.0.0.1:5984
+  SERVER_USERNAME: admin
+  SERVER_PASSWORD: password
+jobs:
+  test:
+    strategy:
+      matrix:
+        java: ['8'] 
+        # java: ['8', '11', '17'] 
+    runs-on: ubuntu-latest
+    services:
+      couchdb:
+        image: apache/couchdb:3
+        env:
+          COUCHDB_USER: ${{ env.SERVER_USERNAME }}
+          COUCHDB_PASSWORD: ${{ env.SERVER_PASSWORD }}
+        options: --name couchdb
+        ports:
+        - 5984:5984
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+    - name: Setup java ${{ matrix.java }}
+      uses: actions/setup-java@v3
+      with:
+        distribution: 'temurin'
+        java-version: ${{ matrix.java }}
+        cache: 'maven'
+        check-latest: false
+    - name: Setup CouchDB
+      shell: bash
+      run: ${GITHUB_WORKSPACE}/scripts/setup_couch.sh
+    - name: Set maven version
+      if: github.event_name == 'create' && startsWith(github.ref, 'refs/tags/')
+      run: mvn versions:set -DnewVersion=${{ github.ref_name }} -DgenerateBackupPoms=false
+    - name: Run tests
+      run: mvn --batch-mode verify
```

Comment:
I'm not opposed to using `batch-mode` here, but I didn't see any failure that seemed to necessitate it, was there a problem?